### PR TITLE
Clean up Makefile variables and comments

### PR DESCRIPTION
**Pull request by mrymsza**
Keep all variables at the top of the file
Remove unused and/or unnecessary variables
Delete unnecessary comments

Fixes https://github.com/k8snetworkplumbingwg/sriov-network-device-plugin/issues/291

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -11,18 +11,22 @@ GOPATH=$(CURDIR)/.gopath
...
 BASE=$(GOPATH)/src/$(REPO_PATH)
-PKGS     = $(or $(PKG),$(shell cd $(BASE) && env GOPATH=$(GOPATH) $(GO) list ./... | grep -v "^$(PACKAGE)/vendor/"))
+PKGS = $(or $(PKG),$(shell cd $(BASE) && env GOPATH=$(GOPATH) go list ./... | grep -v "^$(PACKAGE)/vendor/"))
 GOFILES = $(shell find . -name *.go | grep -vE "(\/vendor\/)|(_test.go)")
-TESTPKGS = $(shell env GOPATH=$(GOPATH) $(GO) list -f '{{ if or .TestGoFiles .XTestGoFiles }}{{ .ImportPath }}{{ end }}' $(PKGS))
-
-export GOPATH
-export GOBIN
-
-# Docker
```
Comment (adrianchiris) [inline Makefile:21]: even after the refactoring i would prefer if we could group the variables according to their related usage, e.g have a comment stating that the following variables are docker related so they are not bunched in a single long list of variables.

File: Makefile [hunk 2]
```diff
@@ -37,11 +41,9 @@ ifdef STATIC
 	LDFLAGS=-a -ldflags '-extldflags \"-static\"'
 endif
 
-# Go tools
-GO      = go
```
Comment (adrianchiris) [inline Makefile:41]: While this is best practice in C/C++ where different compilers are used, i think we only have one compiler in go right ?

So im fine with not having this as a parameter (i also see that other parts of this makefile directly reference go compiler directly)

File: Makefile [hunk 3]
```diff
@@ -134,23 +122,18 @@ lint: | $(BASE) $(GOLINT) ; $(info  running golint...) @ ## Run golint
...
 	 done ; exit $$ret
 
-# Dependency management
 .PHONY: deps-update
 deps-update: ; $(info  updating dependencies...)
-	go mod tidy && go mod vendor
+	@go mod tidy && go mod vendor
 
-# Docker image
-# To pass proxy for Docker invoke it as 'make image HTTP_POXY=http://192.168.0.1:8080'
```
Comment (adrianchiris) [inline Makefile:147]: nit: i think its not half bad to keep this comment.

**Review by adrianchiris — APPROVED** [review]
LGTM Thanks!

**Review by martinkennelly — APPROVED** [review]
Thanks for the changes

**Pull request merged**